### PR TITLE
Fix `React.lazy` and `React.forwardRef` component updates tracking.

### PR DESCRIPTION
**Pull request by XantreDev**
Based on @ywang1724 pr #342. Appreciate for help.

Added tests of reactiveness for `React.lazy` and `React.forwardRef` and made program to pass it. 
Updated pnpm-lock while installation, because ci failed

### CODE REVIEW

**Comment by XantreGodlike** [comment]
@marvinhagemeister @JoviDeCroock Review me please). Its fix for 1.2.1 version, because its most stable one and for now the only one working in react native.
#301 #420

File: pnpm-lock.yaml [hunk 1]
```diff
@@ -1,243 +1,308 @@
-lockfileVersion: 5.4
+lockfileVersion: '6.0'
```
Comment (JoviDeCroock) [inline pnpm-lock.yaml:1]: I don't think adding the lockfile changes is in scope of this PR

**Review by JoviDeCroock — COMMENTED** [review]
Seems to have a lot of merge conflicts and non-related changes in the lockfile

**Comment by rschristian** [comment]
> Its fix for 1.2.1 version, because its most stable one and for now the only one working in react native.

Am I understanding you correctly in that this is based on `1.2.1`, instead of the branch head? Definitely can't merge this into main if so.

Precisely, is it `1.2.1` or `1.2.2`? Should probably be the latter and we can make a branch for you to merge into if needs be.

File: pnpm-lock.yaml (hunk 1 above)
Comment (XantreGodlike) [reply pnpm-lock.yaml:1]: Sorry but without this changes pipelines was failed

**Comment by XantreGodlike** [comment]
I think it should be in another branch, this changes for `1.2.2`

**Comment by XantreGodlike** [comment]
I think we can cherry pick commit related with adding tests for forwardRef and lazy. But other things merge into separate branch.

File: pnpm-lock.yaml (hunk 1 above)
Comment (rschristian) [reply pnpm-lock.yaml:1]: Then we need to update/alter the pipeline or fix it separately.

Comment (XantreGodlike) [reply pnpm-lock.yaml:1]: Should I cherry pick and provide another PR? In which branch i should merge it?

Comment (XantreGodlike) [reply pnpm-lock.yaml:1]: @rschristian What should I do?

Comment (XantreGodlike) [reply pnpm-lock.yaml:1]: @JoviDeCroock

**Comment by XantreGodlike** [comment]
@JoviDeCroock @rschristian what should i do to release 1.2.3 with this fixes? I want to fix this bug, because it causes problems in my project

**Comment by rschristian** [comment]
Sorry for the delay, just been incredibly busy and missed this. I created a new branch from `1.2.2` and switched this to target it. You might want to take a look at the `pnpm.lock` file though, that diff looks odd and unnecessary.

I don't use React and therefore this package, so I can't really comment here. @andrewiggins is probably the best person to comment on this (if he has the time)

**Comment by XantreGodlike** [comment]
#431 Extracted lockfile changes to separate PR

**Comment by rschristian** [comment]
The CI being limited to PRs against `main` is one way to fix the issue 😂

**Comment by aspizu** [comment]
any updates on this?

**Comment by XantreGodlike** [comment]
> any updates on this?

You can use the 2.0.0 version with the babel plugin now

**Pull request closed without merging**